test/integration/policy/policy_test.go

```diff
@@ -0,0 +1,94 @@
+package policy
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-network-policy-agent/test/framework/manifest"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	network "k8s.io/api/networking/v1"
+)
+
+var _ = Describe("Network Policy Test Cases", func() {
+	Context("A network policy is applied against a pod with '.' in its pod name", func() {
+
+		var clientPod *v1.Pod
+		var clientNetworkPolicy *network.NetworkPolicy
+		var podName = "this.pod.has.dots.in.its.name"
+
+		BeforeEach(func() {
+			By("Creating a network policy that denies all egress traffic", func() {
+				clientNetworkPolicy = manifest.NewNetworkPolicyBuilder().
+					Namespace(namespace).
+					Name("deny-all-egress").
+					PodSelector("app", podName).
+					SetPolicyType(false, true).
```

Comment:
I had tried that way but I found that the network policy kept getting set to an `Ingress` type. It turns out that I needed a way to explicitly set the policy to `Egress` since

> ...if you want to write a policy that specifies that no egress is allowed, you must specify a policyTypes value that include "Egress" (since such a policy would not include an egress section and would otherwise default to just [ "Ingress" ]).

from [NetworkPolicySpec](https://pkg.go.dev/k8s.io/api/networking/v1#NetworkPolicySpec)